From e3811487dd1754571d47428fc64362471e220118 Mon Sep 17 00:00:00 2001
From: Jordan Crouse <jcrouse@codeaurora.org>
Date: Wed, 28 Dec 2011 00:39:06 +0100
Subject: [PATCH] msm: kgsl: remove implementation of
 IOCTL_KGSL_DEVICE_REGREAD

This ioctl isn't used by userspace.

Change-Id: If51a949edc6882a92aa925d5c3b2ec961462f7e8
Signed-off-by: Jeremy Gebben <jgebben@codeaurora.org>
---
 drivers/video/msm/gpu/kgsl_adreno205_hc/kgsl.c |   20 --------------------
 1 files changed, 0 insertions(+), 20 deletions(-)

diff --git a/drivers/video/msm/gpu/kgsl_adreno205_hc/kgsl.c b/drivers/video/msm/gpu/kgsl_adreno205_hc/kgsl.c
index 1731915..6e0be10 100644
--- a/drivers/video/msm/gpu/kgsl_adreno205_hc/kgsl.c
+++ b/drivers/video/msm/gpu/kgsl_adreno205_hc/kgsl.c
@@ -832,24 +832,6 @@ static long kgsl_ioctl_device_getproperty(struct kgsl_device_private *dev_priv,
 	return result;
 }
 
-static long kgsl_ioctl_device_regread(struct kgsl_device_private *dev_priv,
-				      unsigned int cmd, void *data)
-{
-	struct kgsl_device_regread *param = data;
-
-	if (param->offsetwords*sizeof(uint32_t) >=
-	    dev_priv->device->regspace.sizebytes) {
-		KGSL_DRV_ERR(dev_priv->device, "invalid offset %d\n",
-			     param->offsetwords);
-		return -ERANGE;
-	}
-
-	kgsl_regread(dev_priv->device, param->offsetwords, &param->value);
-
-	return 0;
-}
-
-
 static long kgsl_ioctl_device_waittimestamp(struct kgsl_device_private
 						*dev_priv, unsigned int cmd,
 						void *data)
@@ -1756,8 +1738,6 @@ typedef long (*kgsl_ioctl_func_t)(struct kgsl_device_private *,
 } kgsl_ioctl_funcs[] = {
 	KGSL_IOCTL_FUNC(IOCTL_KGSL_DEVICE_GETPROPERTY,
 			kgsl_ioctl_device_getproperty, 1),
-	KGSL_IOCTL_FUNC(IOCTL_KGSL_DEVICE_REGREAD,
-			kgsl_ioctl_device_regread, 1),
 	KGSL_IOCTL_FUNC(IOCTL_KGSL_DEVICE_WAITTIMESTAMP,
 			kgsl_ioctl_device_waittimestamp, 1),
 	KGSL_IOCTL_FUNC(IOCTL_KGSL_RINGBUFFER_ISSUEIBCMDS,
-- 
1.7.5.4

